### ISSUE TITLE
please register `WebSocketHandlerAdapter` by default

### ISSUE THREAD

**Issue by joshlong** [spring-projects/spring-boot#16185]
It seems like I almost never need to do anything with `WebSocketHandlerAdapter` when using WebFlux and websockets. I just need to register it, like this: 

```
	@Bean
	SimpleUrlHandlerMapping simpleUrlHandlerMapping(WebSocketHandler wsh) {
		return new SimpleUrlHandlerMapping() {
			{
				setUrlMap(Map.of("/ws/time", wsh));
				setOrder(10);
			}
		};
	}

	@Bean
	WebSocketHandlerAdapter wsha() {
		return new WebSocketHandlerAdapter();
	}

	@Bean
	WebSocketHandler wsh(Timer timer) {
		return session -> session.send(timer.greet().share().map(session::textMessage));
	}
```

I can see why i need to register the `WebSocketHandler`  (that's where my custom websocket logic lives) and I can see why I need to register the `SimpleUrlHandlermapping` bean (that's where my custom routing lives). But why the `WebSocketHandlerAdapter`? Could it be auto-configuration registered based on the presence of websockets on the classpath and subject to a `@ConditionalOnMissingBean` so that users who want to contribute their own can do so? it seems ceremonial. 

thanks in advance..

**Comment by philwebb** [comment]
> Could it be auto-configuration registered based on the presence of websockets on the classpath

If we're talking about `org.springframework.web.reactive.socket.server.support.WebSocketHandlerAdapter` then I'm not sure we have a strong classpath signal that websockets are being used since that class is part of `spring-webflux`.

@joshlong or @rstoyanchev Do you know if a `spring-websocket` jar dependency is also needed? Or do reactive websockets work differently?

**Comment by rstoyanchev** [comment]
No, `spring-websocket` is not need/used for this but we could probably register `WebSocketHandlerAdapter` automatically with the `@EnableWebFlux` setup, and perhaps add some extra convenience methods to `SimpleUrlHandlerMapping` to improve the above sample. @philwebb feel free to transfer to spring-framework.
